### PR TITLE
Move several costing features to base class

### LINKED ISSUE #1581: Routes through the roads with tags "access=private".
Navigation mode for cyclists sometimes builds a route through the roads with tags "access=private". 

For example:
through this factory: https://www.openstreetmap.org/#map=14/47.1603/37.5605

image2

### PR DESCRIPTION
**Pull request by dnesbitt61**
Almost all of the costing methods have similar code for dealing with specific transitions: ferries, gates, country crossing, tolls, alleys. Destination only (private) roads is another, but bicycle costing did not have this and treated destination only roads differently (not as effectively since it was based on a weight factor which highly depends on road length).

This PR consolidates the "base" transition costs in the base DynamicCost class.  The costs and penalties are also moved from the individual costing methods to the base class. Each derived class can provide its own costs and penalties.

fixes #1581 

 - [x] Add tests
 - [x] Review - you must request approval to merge any PR to master
 - [x] Add #fixes with the issue number that this PR addresses
 - [x] Generally use squash merge to rebase and clean comments before merging
 - [x] Update the [changelog](CHANGELOG.md)

### CODE REVIEW

**Comment by gknisely** [comment]
Are you going to run the RAD tests against this branch?

**Comment by dgearhart** [comment]
code looks good - how do test routes look?

**Comment by dnesbitt61** [comment]
auto routes match...after "fixing" the toll cost with the last commit.

**Comment by dnesbitt61** [comment]
still working on tests...

File: src/sif/bicyclecost.cc [hunk 1]
```diff
@@ -1136,56 +1042,89 @@ void testBicycleCostParams() {
...
+  distributor.reset(make_distributor_from_range(kFerryCostRange));
+  for (unsigned i = 0; i < testIterations; ++i) {
+    ctorTester.reset(make_bicyclecost_from_json("ferry_cost", (*distributor)(generator)));
+    if (ctorTester->ferry_transition_cost_.secs < kFerryCostRange.min ||
+        ctorTester->ferry_transition_cost_.secs > kFerryCostRange.max) {
+      throw std::runtime_error("ferry_cost_ is not within it's range");
+    }
+  }
+
+  /**
```
Comment (gknisely) [inline src/sif/bicyclecost.cc]: @dnesbitt61 is this supposed to be commented out?

**Comment by dnesbitt61** [comment]
Note that some nline costing tests are currently disabled. Some input parameters are validated for range but then are used tocreate a separate costing parameter. So the input parameter doesn't need to be a class member.These seem to be validated somewhat in the parse_request test.

**Pull request merged**